### ISSUE TITLE
Search box on Quick Order page doesn't support keyboard navigation 

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#14187]
**Describe the bug**
- Tabbing away from the suggestions listbox leaves it open.
- X button is not keyboard accessible, therefore search string can't be cleared.
- Suggestions listbox should not receive focus as a group.
- While the Suggestions listbox is open and focused, and trying to close it with ESC, at some point, the focus jumps up to the skippers

**Tell us the version of Spartacus**
- Library version: 4.2.RC0

**To Reproduce**
1. Go to the Quick Order page and start tabbing to reach the search box.
2. Once in the search box type something to trigger suggestions. E.g. "cameras"
3. Tab again to reach the "X" button, to try and clear the search string.
4. Notice the suggestion listbox getting focused as a group.
4. Tab again to move out of the suggestions box.
5. See both errors.

**Expected behavior**
- Tabbing away from the suggestions listbox should close it.
- X button should be keyboard accessible, in order to clear the search string.
- **Same behaviour as in Main Search.**

**Screenshots**

- Suggestions listbox receveing focus as group, skipping the "X" button:

![keyboard-nav](https://user-images.githubusercontent.com/15113219/139757331-1c158e2f-a50b-48f8-a12e-7a8d1d30dd5e.jpg)
- Suggestions listbox stays open while tabbing away:

![stays-open](https://user-images.githubusercontent.com/15113219/139757472-2b60ec0c-7421-404f-a03b-7bc99f69de62.jpg)
![hidden-focus-ring](https://user-images.githubusercontent.com/15113219/139757480-5e024edf-03aa-40b1-a30d-7531673d4649.jpg)

- Hitting ESC while focusing the suggestions listbox makes the focus jump to Skippers.

![Screen Shot 2021-11-01 at 8 03 32 PM](https://user-images.githubusercontent.com/15113219/139758730-a7566942-430c-4e38-9559-6b43a5f10b42.png)

**Comment by giancorderoortiz** [comment]
please merge fix info 4.2.x branch

**Comment by dydome** [comment]
@giancorderoortiz it was merged https://github.com/SAP/spartacus/pull/14249
